astropy/table/column.py

```diff
@@ -292,6 +291,13 @@ def copy(self, order='C', data=None, copy_data=True):
 
         out = data.view(self.__class__)
         out.__array_finalize__(self)
+
+        # If there is meta on the original column then deepcopy (since "copy" of column
+        # implies complete independence from original).  __array_finalize__ will have already
+        # made a light copy.  I'm not sure how to avoid that initial light copy.
```

Comment:
It is not completely unreasonable to just remove `__array_finalize__` and just call `_copy_attrs` - but this may go wrong for `MaskedColumn`. Alternatively, perhaps this whole thing should instead just go via class creation (i.e., something like `self.__class__(..., copy=...)`?

Maybe for another PR, though...

Reply:
Agreed that this needs investigation, but definitely prefer another PR.

Reply:
OK, maybe just add "TODO: investigate if going through class creation is easier/faster."?